samples/subsys/shell/shell_module/src/main.c

```diff
@@ -13,6 +13,12 @@
 #include <zephyr/usb/usb_device.h>
 #include <ctype.h>
 
+#ifdef CONFIG_ARCH_POSIX
+#include <unistd.h>
+#else
+#include <zephyr/posix/unistd.h>
+#endif
```

Comment:
Having to do this everywhere is kind of yucky. Can't we add zephyr/posix to the include path if the posix subsystem is enabled or so?

Reply:
It sure is!

Part of the LTSv3 POSIX API Roadmap is to eliminate the mutually-exclusiveness of CONFIG_ARCH_POSIX and CONFIG_POSIX_API.

So although this seems like technical debt, it's actually part of the process of fixing several years worth of other technical debt / neglect.

Eventually, to use POSIX APIs, we will need to simply use CONFIG_POSIX_API=y and use the actual standard paths.